### PR TITLE
pkg/terminal: fix handling list colors via config

### PR DESCRIPTION
**Pull request by derekparker**
Parsing the source list color when set during a Delve debug session resulted in unexpected errors. Additionally the changes were not reflected in the current session, forcing the user to save the config and start a new session. This patch fixes those issues.

Fixes https://github.com/go-delve/delve/issues/3238

### CODE REVIEW

**Comment by derekparker** [comment]
@thockin can you test this out?

File: pkg/config/split.go [hunk 1]
```diff
@@ -84,7 +84,29 @@ func ConfigureSetSimple(rest string, cfgname string, field reflect.Value) error
 			v := rest == "true"
 			return reflect.ValueOf(&v), nil
 		case reflect.String:
+			var err error
+			rest, err = strconv.Unquote(rest)
```
Comment (thockin) [inline pkg/config/split.go]: Thios will fail if it is not quoted, which will break anyone who relies on YAML sloppy strings or even contextual interp of numbers as strings.  Maybe not important for these fields, but it IS a breaking change.

see https://github.com/go-delve/delve/pull/3239/files

File: pkg/config/split.go [hunk 2]
```diff
@@ -84,7 +84,29 @@ func ConfigureSetSimple(rest string, cfgname string, field reflect.Value) error
...
+			// We special case this particular configuration key because historically we accept both a numerical value and a string value for it.
+			if cfgname == "source-list-line-color" {
+				n, err := strconv.Atoi(rest)
+				if err == nil {
+					if n < 0 {
+						return reflect.ValueOf(nil), fmt.Errorf("argument to %q must be a number greater than zero", cfgname)
+					}
+					return reflect.ValueOf(&n), nil
+				}
+				rest, err = strconv.Unquote(rest)
```
Comment (thockin) [inline pkg/config/split.go]: same

File: pkg/terminal/terminal.go [hunk 3]
```diff
@@ -141,6 +141,34 @@ func New(client service.Client, conf *config.Config) *Term {
 	return t
 }
 
+func (t *Term) updateColorScheme() {
```
Comment (thockin) [inline pkg/terminal/terminal.go:144]: remove the dup block in `New()` and call this?

**Review by thockin — COMMENTED** [review]
It works for me

File: pkg/config/split.go (hunk 1 above)
Comment (derekparker) [reply pkg/config/split.go]: Absolutely, you're right. Was working on some follow-up fixes / refactoring after submitting for testing. They are included in this PR now.

File: pkg/config/split.go [hunk 4]
```diff
@@ -84,7 +84,28 @@ func ConfigureSetSimple(rest string, cfgname string, field reflect.Value) error
 			v := rest == "true"
 			return reflect.ValueOf(&v), nil
 		case reflect.String:
+			unquoted, err := strconv.Unquote(rest)
```
Comment (thockin) [inline pkg/config/split.go:87]: How do you want to handle cases like `"forgot the last quote` ?

My PR handles it as an error (clearly they meant to specify a quoted string).  This PR will treat it as a literal (which is more back-compatible, but less correct.

You decide - the rest LGTM :)

**Comment by derekparker** [comment]
Fixed a few failing tests due to changes in how config output is presented, but that's really a minor change inconsequential to the overall change. With that in mind, I may still back it out in case there are clients that depend on the current format. @suzmue @polinasok @hyangah there was a DAP test which failed, does the VSCode DAP client depend on parsing the config format in a specific format at all? If so I can revert that change.

**Review by aarzilli — APPROVED** [review]
LGTM

**Pull request merged**